conda/petsc/repeat_build.sh

```diff
@@ -0,0 +1,25 @@
+source $SRC_DIR/configure_petsc.sh
+configure_petsc \
+    --COPTFLAGS=-O3 \
+    --CXXOPTFLAGS=-O3 \
+    --FOPTFLAGS=-O3 \
```

Comment:
How'd we come up with these?

Reply:
`-O3` is a logical choice. I'm just commenting because we do `-O2` in libmesh/MOOSE and IIRC @roystgnr may have said we see now benefits from `-O3` or even a slow-down(?)

Reply:
When we borrowed sections from petsc-feedstock on conda-forge when we first created this package.

Ref: https://github.com/conda-forge/petsc-feedstock/blob/main/recipe/build.sh

Reply:
Yeah pretty cool!